src/pybind/mgr/cephadm/module.py

```diff
@@ -561,6 +562,8 @@ def __init__(self, *args, **kwargs):
         self.cache = HostCache(self)
         self.cache.load()
 
+        self.service_spec_store = PersistentStoreDict(self, 'service_spec')
```

Comment:
You have this top-level global instance,but you only use it for 'ls'... and I think that only works because your per-service-type ones are nested beneath it.  I suggest just putting everything in this dict and avoiding ad hoc instantiations of PersistentStoreDict sprinkled everywhere